cpp/CMakeLists.txt

```diff
@@ -37,6 +37,13 @@ rapids_cmake_write_version_file(include/rapidsmpf/version_config.hpp)
 # Set a default build type if none was specified
 rapids_cmake_build_type(Release)
 
+# Set RAPIDSMPF_DEBUG default based on build type
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(RAPIDSMPF_DEBUG_DEFAULT ON)
+else()
+  set(RAPIDSMPF_DEBUG_DEFAULT OFF)
+endif()
+
```

Comment:
I would prefer that the default is always off, even when CMake build type is debug, thus requiring you to specify it manually via `--cmake-args`, for example. The reason for that is we are more likely to build in debug mode to do general debugging than to require the extended NVTX annotations.

Furthermore, I think it makes more sense to have a different name, debug invokes a meaning of trying to resolve a problem, whereas I think this is more about performance validation/investigation. How do you feel about `RAPIDSMPF_NVTX_DETAILED`?

Reply:
A `DEBUG` mode macro came up in several discussions. So, I would like this to be used for not just nvtx annotations.

Reply:
I think `DEBUG` is a more specific term IMO, I'd prefer `RAPIDSMPF_DETAIL`. In any case, I think we should only enable it when explicitly specified, and not just depend on `CMAKE_BUILD_TYPE=Debug` to enable it by default.